fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java

```diff
@@ -152,46 +99,80 @@ public boolean onOptionsItemSelected(MenuItem item) {
     return super.onOptionsItemSelected(item);
   }
 
-  public void setupNavigationBar() {
-    BottomNavigationView bottomNavigationView = findViewById(R.id.social_media_nav_bar);
-    bottomNavigationView.setOnItemSelectedListener(
+  private void setupNavigationBar() {
+    navbar = findViewById(R.id.social_media_nav_bar);
+    navbar.setOnItemSelectedListener(
         item -> {
           int itemId = item.getItemId();
-          if (itemId == R.id.social_media_home_menu) {
-            mViewModel.openHome();
-          } else if (itemId == R.id.social_media_search_menu) {
-            mViewModel.openSearch();
-          } else if (itemId == R.id.social_media_following_menu) {
-            mViewModel.openFollowing();
-          } else if (itemId == R.id.social_media_profile_menu) {
-            mViewModel.openProfile();
+          if (itemId != mViewModel.getCurrentSelectedItem().getValue()) {
+            mViewModel.setCurrentSelectedItem(itemId);
```

Comment:
The problem is that it would be executed twice when the user select a menu item themself. It would not be problematic most of the time I imagine but it's something we should avoid I think.
I'll add the comment I put in digital cash for the same check

Reply:
Oh ok, tanks for the clarification